src/overlays/actors/ovl_En_Nb/z_en_nb.c

```diff
@@ -21,255 +169,1401 @@ const ActorInit En_Nb_InitVars = {
+    this->action = NB_KNEEL;
+    this->drawMode = NB_DRAW_KNEEL;
+    this->actor.shape.unk_14 = 0xFF;
+}
+
+void EnNb_CheckIfKneeling(EnNb* this, s32 animFinished) {
+    AnimationHeader* animation = &D_060046A8;
+    if (animFinished) {
+        SkelAnime_ChangeAnim(&this->skelAnime, animation, 1.0f, 0.0f,
+                             SkelAnime_GetFrameCount(&animation->genericHeader), 0, 0.0f);
+        this->drawMode = NB_DRAW_KNEEL;
+    }
+}
+
+void EnNb_SetupLookRight(EnNb* this) {
+    AnimationHeader* animation = &D_06003954;
+    f32 frames = SkelAnime_GetFrameCount(&animation->genericHeader);
+    SkelAnime_ChangeAnim(&this->skelAnime, animation, 1.0f, 0.0f, frames, 2, -8.0f);
+    this->action = NB_LOOK_RIGHT;
+    this->drawMode = NB_DRAW_DEFAULT;
+    this->actor.shape.unk_14 = 0xFF;
+}
+
+void EnNb_CheckIfLookingRight(EnNb* this, s32 animFinished) {
+    AnimationHeader* animation = &D_06004030;
+    if (animFinished) {
+        SkelAnime_ChangeAnim(&this->skelAnime, animation, 1.0f, 0.0f,
+                             SkelAnime_GetFrameCount(&animation->genericHeader), 0, 0.0f);
+        this->drawMode = NB_DRAW_LOOK_DIRECTION;
+    }
+}
+
+void EnNb_SetupLookLeft(EnNb* this) {
+    AnimationHeader* animation = &D_06002DBC;
+    f32 frames = SkelAnime_GetFrameCount(&animation->genericHeader);
+    SkelAnime_ChangeAnim(&this->skelAnime, animation, 1.0f, 0.0f, frames, 2, -8.0f);
+    this->action = NB_LOOK_LEFT;
+    this->drawMode = NB_DRAW_LOOK_DIRECTION;
+    this->actor.shape.unk_14 = 0xFF;
+}
+
+void EnNb_CheckIfLookLeft(EnNb* this, s32 animFinished) {
+    AnimationHeader* animation = &D_060035A8;
+    if (animFinished) {
+        SkelAnime_ChangeAnim(&this->skelAnime, animation, 1.0f, 0.0f,
+                             SkelAnime_GetFrameCount(&animation->genericHeader), 0, 0.0f);
+    }
+}
+
+void EnNb_SetupDemo6KInConfrontation(EnNb* this, GlobalContext* globalCtx, s32 animFinished) {
+    if (!this->flag && animFinished) {
+        EnNb_InitDemo6KInConfrontation(this, globalCtx);
+        this->flag = 1;
+    }
+}
+
+void EnNb_SetupRun(EnNb* this) {
+    AnimationHeader* animation = &D_06006320;
+    f32 frames = SkelAnime_GetFrameCount(&animation->genericHeader);
+    SkelAnime_ChangeAnim(&this->skelAnime, animation, 1.0f, 0.0f, frames, 2, -8.0f);
+    this->action = NB_RUN;
+    this->drawMode = NB_DRAW_LOOK_DIRECTION;
+    this->actor.shape.unk_14 = 0xFF;
+}
+
+void EnNb_SetupConfrontationDestroy(EnNb* this) {
+    this->action = NB_CONFRONTATION_DESTROYED;
+    this->drawMode = NB_DRAW_NOTHING;
+    this->actor.shape.unk_14 = 0;
+}
+
+void EnNb_CheckConfrontationCsMode(EnNb* this, GlobalContext* globalCtx) {
+    CsCmdActorAction* csCmdNPCAction;
+    s32 csAction;
+    s32 csActionMinusOne;
+    s32 unk_28C;
+
+    csCmdNPCAction = EnNb_GetNpcCsAction(globalCtx, 1);
+    if (csCmdNPCAction != 0) {
+        csAction = csCmdNPCAction->action;
+        csActionMinusOne = csAction - 1;
```

Comment:
is `csActionMinusOne` real or can you use `csAction` and bump each switch case by 1 instead?